src/main/kotlin/autodiff/Expression.kt

```diff
@@ -10,21 +10,17 @@ abstract class Expression {
 
     abstract fun getVariables(): Set<Variable>
 
-    operator fun plus(expression : Expression) : Expression {
-        return Sum(this, expression)
-    }
+    operator fun plus(expression : Expression) : Expression { return Sum(this, expression) }
```

Comment:
Can be done with equals expressions. `operator fun plus(expression: Expression) = Sum(this, expression)`